File: src/main/java/com/spending/model/Registry.java
package com.spending.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Builder
@ToString
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = {"date", "description", "value"})
@Data
@Document
public class Registry {

    @Id
    private String id;
    private String date;
    private String description;
    private String type;
    private Double value;
    private String category;

}

File: front/src/app/registry/registry.ts
import { Type } from '../type/type';
import { Category } from '../category/category';
export class Registry {

    id: string;
    date: string;
    description: string;
    type: Type = new Type();
    value: number;
    category: Category = new Category();

    constructor(values: Object = {}) {
      // Constructor initialization
      Object.assign(this, values);
    }
}
